### PR TITLE
Sysmon bugfix

### PR DESCRIPTION
**Pull request by ashwin-patil**
Fixes #

## Proposed Changes

  - Fix for sysmon parser for Event ID 3 as per Issue #387 
  - Update links in readme for Hunting queries

### CODE REVIEW

File: Parsers/SysmonParser.txt [hunk 1]
```diff
@@ -64,7 +64,6 @@ User = EventDetail.[5].["#text"], Protocol = EventDetail.[6].["#text"], Initiate
 SourceHostname = EventDetail.[10].["#text"], SourcePort = EventDetail.[11].["#text"], SourcePortName = EventDetail.[12].["#text"], DestinationIsIpv6 = EventDetail.[13].["#text"], 
 DestinationIp = EventDetail.[14].["#text"], DestinationHostname = EventDetail.[15].["#text"], DestinationPort = EventDetail.[16].["#text"], DestinationPortName = EventDetail.[17].["#text"]
 | parse RuleName with * 'technique_id=' TechniqueId ',' * 'technique_name=' TechniqueName 
-| parse Hashes with * 'SHA1=' SHA1 ',' * 'MD5=' MD5 ',' * 'SHA256=' SHA256 ',' * 'IMPHASH=' IMPHASH 
 | project-away EventDetail, RuleName, Hashes
```
Comment (ssi0202) [inline Parsers/SysmonParser.txt]: i think you need to lose the "Hashes" in line project away line too, just below

Comment (ashwin-patil) [reply Parsers/SysmonParser.txt]: removed

File: Parsers/SysmonParser.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // KQL Sysmon Event Parser
-// Last Updated Date: Sept 4, 2019
+// Last Updated Date: Nov 25, 2019
 // Sysmon Version : 10.2.0.0
```
Comment (shainw) [inline Parsers/SysmonParser.txt]: Change Sysmon Version to 10.41, also, lets add in the Schema Version of the lastest - schemaversion="4.22"

**Comment by ssi0202** [comment]
will this pull fix all issues, was it only eventID 3 and 7 that had issues after all ?

File: Parsers/SysmonParser.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
 //
 // Sysmon Instructions:
```
Comment (shainw) [inline Parsers/SysmonParser.txt]: After thinking about this further, I think we will need multiple parser versions as changes like this occur, but where older versions of sysmon are still deployed and producing data with the older schema.  Let's look at creating a new file with a version in the file name.

File: Parsers/Sysmon-v9-Parser.txt [hunk 4]
```diff
@@ -1,27 +1,26 @@
 // KQL Sysmon Event Parser
-// Last Updated Date: Nov 25, 2019
-// Sysmon Version : 10.2.0.0
+// Last Updated Date: Dec 9, 2019
+// Sysmon Binary Version : 9.10 , Schema Version: 4.21
```
Comment (shainw) [inline Parsers/Sysmon-v9-Parser.txt]: Lets leave in the Sysmon Version, same as I suggested above.  Helps make sure users understand what version this supports.

File: Hunting Queries/AzureDiagnostics/CriticalPortsOpened.yaml [hunk 5]
```diff
@@ -38,4 +38,4 @@ query: |
   | where type_s !~ "block"
   | order by TimeGenerated desc
   | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), count() by OperationName, systemId_g, vnetResourceGuid_g, subnetPrefix_s, macAddress_s, primaryIPv4Address_s, ruleName_s,
-direction_s, priority_d, type_s, conditions_destinationIP_s, conditions_destinationPortRange_s, conditions_sourceIP_s, conditions_sourcePortRange_s, ResourceId
+   direction_s, priority_d, type_s, conditions_destinationIP_s, conditions_destinationPortRange_s, conditions_sourceIP_s, conditions_sourcePortRange_s, ResourceId
```
Comment (shainw) [inline Hunting Queries/AzureDiagnostics/CriticalPortsOpened.yaml]: Not sure anything changed here?

File: Parsers/Sysmon-v9-Parser.txt (hunk 4 above)
Comment (ashwin-patil) [reply Parsers/Sysmon-v9-Parser.txt]: Added details and renamed files

File: Parsers/Sysmon-v10-Parser.txt [hunk 6]
```diff
@@ -0,0 +1,274 @@
+// KQL Sysmon Event Parser
+// Last Updated Date: Dec 9,  2019
+// Sysmon Binary Version : 9.10 , Schema Version: 4.21
```
[the remark this answers is not in the document]
Comment (ashwin-patil) [reply Parsers/Sysmon-v10-Parser.txt]: Added details and renamed files

File: Hunting Queries/CustomLogs/AWSBucketAPILogs-SuspiciousDataAccessToS3BucketsfromUnknownIP.yaml [hunk 7]
```diff
@@ -16,20 +16,20 @@ relevantTechniques:
...
-let starttime = 14d;
-let midtime = 2d;
-let endtime = 1d;
-AWSS3BucketAPILogParsed 
-| where EventTime >= ago(endtime)
-| where EventName in (EventNameList)
-| project EventTime, EventSource,EventName, SourceIPAddress, UserIdentityType, UserIdentityArn, UserIdentityUserName, BucketName, Host, AuthenticationMethod, SessionMfaAuthenticated, SessionUserName, Key
-| join kind=leftanti
-(
+    let EventNameList = dynamic(["ListBucket","ListObjects","GetObject"]);
```
[the remark this answers is not in the document]
Comment (ashwin-patil) [reply Hunting Queries/CustomLogs/AWSBucketAPILogs-SuspiciousDataAccessToS3BucketsfromUnknownIP.yaml]: corrected

File: Hunting Queries/CustomLogs/AWSBucketAPILogs-SuspiciousCredentialTokenAccessOfValid_IAM_Roles.yaml [hunk 8]
```diff
@@ -17,20 +17,20 @@ relevantTechniques:
...
-| join kind= leftanti (
-AWSCloudTrail
-| where TimeGenerated  between (ago(starttime)..ago(midtime))
-| where EventName == "AssumeRole" | extend RoleArn = tostring(parse_json(RequestParameters).roleArn)
-| project TimeGenerated, EventSource, EventName, UserIdentityType, UserIdentityInvokedBy , SourceIpAddress, RoleArn
-) on RoleArn, UserIdentityInvokedBy
-| summarize EventCount = count(), StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated) by RoleArn, EventSource, EventName, UserIdentityType, UserIdentityInvokedBy, SourceIpAddress
-| extend timestamp = StartTimeUtc, IPCustomEntity = SourceIpAddress, AccountCustomEntity = tostring(split(RoleArn, "/")[1])
\ No newline at end of file
+    let starttime = 14d;
```
[the remark this answers is not in the document]
Comment (ashwin-patil) [reply Hunting Queries/CustomLogs/AWSBucketAPILogs-SuspiciousCredentialTokenAccessOfValid_IAM_Roles.yaml]: corrected. Also noticed this should be under AWS Cloudtrail so moved there with changes in connector details

File: Hunting Queries/CustomLogs/AWSBucketAPILogs-S3BucketDataTransferTimeSeriesAnomaly.yaml [hunk 9]
```diff
@@ -22,29 +22,29 @@ relevantTechniques:
...
-// Use the time series data prepared in previous step with time series aomaly function to generate baseline pattern and flag the outlier based on scorethreshold value.
-let TimeSeriesAlerts = TimeSeriesData
-  | extend (anomalies, score, baseline) = series_decompose_anomalies(Total, scorethreshold, -1, 'linefit')
-  | mv-expand Total to typeof(double), EventTime to typeof(datetime), anomalies to typeof(double), score to typeof(double), baseline to typeof(long)
-  | where anomalies > 0
-  | project EventTime, Total, baseline, anomalies, score;
-// Joining the flagged outlier from the previous step with the original dataset to present contextual information during the anomalyhour to analysts to conduct investigation or informed decistions.
-TimeSeriesAlerts
-  | join (
+    let starttime = 14d;
```
[the remark this answers is not in the document]
Comment (ashwin-patil) [reply Hunting Queries/CustomLogs/AWSBucketAPILogs-S3BucketDataTransferTimeSeriesAnomaly.yaml]: corrected indentation

File: Hunting Queries/AzureDiagnostics/CriticalPortsOpened.yaml (hunk 5 above)
Comment (ashwin-patil) [reply Hunting Queries/AzureDiagnostics/CriticalPortsOpened.yaml]: indentation added in project on last line. no other change

**Pull request merged**